elements/token-list.es

```diff
@@ -2,89 +2,79 @@ class TokenList {
 
   constructor (node) {
 
-    const
-      tokenize = node =>
-        (node.text = node.textContent)
+    this
+      .sift (node)
+      .map  (this.tokenize, this)
+  }
 
-        && node.textContent
-            .match (/{(\w+|#)}/g)
-              .map (symbol => symbol.match (/(\w+|#)/g) [0])
-              .map (insert (node))
+  tokenize (node) {
 
-    , insert = token =>
+    const
+      insert = node =>
         symbol =>
-          (this [symbol] = this [symbol] || [])
-            && this [symbol].push (token)
+          (this [symbol] = this [symbol] || []).push (node)
 
-    this
-      .sift (node)
-      .map  (tokenize)
+    void (node.text = node.textContent)
+      .match (/([^{]*?)\w(?=\})/g)
+      .map (insert (node))
   }
 
-
   sift (node) {
 
     const
       nodes = []
+    , expression = /{(\w+|#)}/
 
     , visit = node =>
         node.nodeType === Node.TEXT_NODE
           ? TEXT_NODE (node)
-            && NodeFilter.FILTER_ACCEPT // Accept TEXT_NODEs
-
           : ELEMENT_NODE (node.attributes)
-            && NodeFilter.FILTER_REJECT // reject ELEMENT_NODEs
+        && NodeFilter.FILTER_REJECT // We don't need 'em
 
     , TEXT_NODE = node =>
-        /{(\w+|#)}/.test (node.textContent)
+        expression.test (node.textContent)
+        && nodes.push (node)
 
-    , ELEMENT_NODE = attributes =>
+    , ELEMENT_NODE = attrs =>
         Array
-          .from (attributes)
-          .filter (attr => /{(\w+|#)}/g.test (attr.value))
-          .map (attribute => nodes.push (attribute))
+          .from (attrs)
+          .map  (attr => expression.test (attr.value) && nodes.push (attr))
 
     , walker =
         document.createNodeIterator
           (node, NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_TEXT, visit)
           // by default breaks on template YAY! 🎉
 
-    while
-      (node = walker.nextNode ())
-        nodes.push (node)
+    while (walker.nextNode ()) 0 // Walk all nodes and do nothing.
```

Comment:
@RobertChristopher Well because we don't need to do anything with the nodes. The `TreeWalker` is doing everything for us. All we are doing is `.push`ing the nodes that have tokens in them during the TreeWalk. `walker.nextNode ()` fires off the treewalker. And each node gets `visit`ed. If the node is a candidate for tokenization then the node simply gets pushed onto a queue.

The `0` is because the `while` operation has to do SOMETHING. (Even if it's nothing`). 
`0` is just one less byte than `{ }` hehe.